build/windows/install_buildenv.bat

```diff
@@ -10,30 +10,46 @@ SET WINLIB_NAME=%2
 REM The local directory to cache build environments in.
 SET WINLIBS_PATH=%3
 
-REM Install build env base dir
-IF NOT EXIST "%WINLIBS_PATH%" MKDIR "%WINLIBS_PATH%"
-
-where /q 7za
-IF ERRORLEVEL 1 (
-  where /q 7z
-  if ERRORLEVEL 1 (
-    echo Could not find 7z or 7za on the path.
-    exit /b
+IF NOT EXIST "%WINLIBS_PATH%\%WINLIB_NAME%" (
+
+  REM Install build env base dir
+  IF NOT EXIST "%WINLIBS_PATH%" MKDIR "%WINLIBS_PATH%"
+
+  where /q 7za
+  IF ERRORLEVEL 1 (
+    where /q 7z
+    if ERRORLEVEL 1 (
+      where /q powershell
+      if ERRORLEVEL 1 (
+         echo Could not find 7z or 7za on the path, and powershell is not available. 
+         echo Either install http://7-zip.org
+         echo or download %BASEURL%/%WINLIB_NAME%.zip
+         echo yourself and unzip it at %WINLIBS_PATH%
+         ENDLOCAL && EXIT /b 1
+      ) else (
+        set POWERSHELL=powershell
+      )
+    ) else (
+      set ZIP=7z
+    )
   ) else (
-    set ZIP=7z
+    set ZIP=7za
   )
-) else (
-  set ZIP=7za
-)
 
-IF NOT EXIST "%WINLIBS_PATH%\%WINLIB_NAME%" (
   echo Installing winlib %WINLIB_NAME%.
   CD "%WINLIBS_PATH%"
   echo ..Downloading %WINLIB_NAME%.zip
   curl -fsS -L -o %WINLIB_NAME%.zip %BASEURL%/%WINLIB_NAME%.zip
+  IF ERRORLEVEL 1 ENDLOCAL && EXIT /b 1
   REM TODO(XXX) check fingerprint on zip file.
   echo ..unzipping %WINLIB_NAME%.zip
-  %ZIP% x %WINLIB_NAME%.zip
+  IF "%POWERSHELL%" == "powershell" (
```

Comment:
This wasn't working without `EnableDelayedLocalExpansion`. I fixed that, and also made a few other changes:

- Don't download if the ZIP already exists. My internet is relatively slow, so I keep a copy around and put it on the build VM as needed.
- Also, if curl isn't on the system, exit and give info on where to download and place the ZIP file

Here's my commit with those changes if you want to merge it into your PR
https://github.com/mixxxdj/mixxx/commit/edeb4cd46974ce6fb8e6ace00161a8092fad357d

Reply:
Do you want to compare the mtime on the ZIP while you're at it?